bundles/org.openhab.binding.intesis/src/main/java/org/openhab/binding/intesis/internal/handler/IntesisHomeHandler.java

```diff
@@ -0,0 +1,552 @@
+                    value = command.equals(OnOffType.OFF) ? 0 : 1;
+                    break;
+                case CHANNEL_TYPE_MODE:
+                    uid = 2;
+                    value = IntesisHomeModeEnum.valueOf(command.toString()).getMode();
+                    break;
+                case CHANNEL_TYPE_FANSPEED:
+                    uid = 4;
+                    if (("AUTO").equals(command.toString())) {
+                        value = 0;
+                    } else {
+                        value = Integer.parseInt(command.toString());
+                    }
+                    break;
+                case CHANNEL_TYPE_VANESUD:
+                    uid = 5;
+                    switch (command.toString()) {
+                        case "AUTO":
+                            value = 0;
+                            break;
+                        case "1":
+                        case "2":
+                        case "3":
+                        case "4":
+                        case "5":
+                        case "6":
+                        case "7":
+                        case "8":
+                        case "9":
+                            value = Integer.parseInt(command.toString());
+                            break;
+                        case "SWING":
+                            value = 10;
+                            break;
+                        case "SWIRL":
+                            value = 11;
+                            break;
+                        case "WIDE":
+                            value = 12;
+                            break;
+                    }
+                    break;
+                case CHANNEL_TYPE_VANESLR:
+                    uid = 6;
+                    switch (command.toString()) {
+                        case "AUTO":
+                            value = 0;
+                            break;
+                        case "1":
+                        case "2":
+                        case "3":
+                        case "4":
+                        case "5":
+                        case "6":
+                        case "7":
+                        case "8":
+                        case "9":
+                            value = Integer.parseInt(command.toString());
+                            break;
+                        case "SWING":
+                            value = 10;
+                            break;
+                        case "SWIRL":
+                            value = 11;
+                            break;
+                        case "WIDE":
+                            value = 12;
+                            break;
+                    }
+                    break;
+                case CHANNEL_TYPE_TARGETTEMP:
+                    uid = 9;
+                    if (command instanceof QuantityType) {
+                        value = ((QuantityType) command).toUnit(SIUnits.CELSIUS).intValue() * 10;
```

Comment:
`toUnit()` is Nullable. Please add a null check.

Reply:
Could you help me with this ??? Can't figure out how to detect if toUnit(Unit) can be checked here.

Reply:
Solved and changed